### PR TITLE
[#4] make compatible with GraalVM

### PR DESCRIPTION
**Pull request by borkdude**
See #4.

### CODE REVIEW

File: src/bencode/core.clj [hunk 1]
```diff
@@ -314,14 +314,15 @@
   'namespace/name'."
   (fn [_output thing]
     (cond
-      (instance? (RT/classForName "[B") thing) :bytes
+      (nil? thing) :list
+      (-> thing class .getComponentType (= Byte/TYPE)) :bytes
```
Comment (bbatsov) [inline src/bencode/core.clj:319]: Might be a good idea to add a comment here we've borrowed this from Clojure 1.9.

File: src/bencode/core.clj [hunk 2]
```diff
@@ -314,14 +314,15 @@
...
       (instance? InputStream thing) :input-stream
       (integer? thing) :integer
       (string? thing)  :string
       (symbol? thing)  :named
       (keyword? thing) :named
       (map? thing)     :map
-      (or (nil? thing) (coll? thing) (.isArray (class thing))) :list
+      (or (coll? thing) (.isArray (class thing))) :list
```
Comment (bbatsov) [inline src/bencode/core.clj:326]: This change is unrelated, so probably it should be in a different commit, so people won't think it's something to do with GraalVM.

Comment (borkdude) [reply src/bencode/core.clj:326]: That change is not unrelated though. As explained, the nil check was moved up, since we needed that in the check for bytes.

File: src/bencode/core.clj (hunk 1 above)
Comment (borkdude) [reply src/bencode/core.clj:319]: Done.

**Comment by bbatsov** [comment]
Thanks!
